osu.Game/Overlays/BeatmapSet/BeatmapSetBadgePill.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class BeatmapSetBadgePill : CircularContainer
+    {
+        public BeatmapSetBadgePill()
+        {
+            AutoSizeAxes = Axes.Both;
+            Masking = true;
+        }
+
+        [BackgroundDependencyLoader(true)]
+        private void load([CanBeNull] OsuColour colours, [CanBeNull] OverlayColourProvider colourProvider)
```

Comment:
Allowing `OsuColour` nullability is a bit overkill imo. No other place does this, it's expected to be always present.

Reply:
It was not annotated with [CanBeNull] but because BackgroundDependencyLoader attribute does not support one dependency to be nullable but others to be non-nullable OsuColour can be null in theory.
I thought it is better to make that explicit than leaving a potential NRE.

Reply:
I'm aware, but other places don't really check or care about that either. If this is a worry then I'd rather have this as an assertion myself.